src/EventStore.ClusterNode/Program.cs

```diff
@@ -167,8 +167,40 @@ internal static class Program {
 					Application.Exit(0, "Cancelled.");
 				};
 
-				var metricsConfiguration = MetricsConfiguration.FromFile();
-				using (var hostedService = new ClusterVNodeHostedService(options, certificateProvider, metricsConfiguration.Get<MetricsConfiguration>())) {
+				// Create a single IConfiguration object that contains the whole configuration, including 
+				// plugin configuration. We will add it to the DI and make it available to the plugins.
+				//
+				// Three json files are loaded explicitly for backwards compatibility
+				// - metricsconfig.json needs loading into the EventStore:Metrics section.
+				// - kestrelsettings.json is not located in a config/ directory
+				// - logconfig.json is not located in a config/ directory
+				var configuration = new ConfigurationBuilder()
+					.AddSection(SectionNames.EventStore, eventStoreSection => eventStoreSection
+						.AddSection(SectionNames.Metrics, metricsSection =>
+							metricsSection.AddConfigFile("metricsconfig.json")))
+
+					// The other config files are added to the root, and must put themselves in the appropriate sections
+					.AddConfigFile("kestrelsettings.json", optional: true, reloadOnChange: true)
+					.AddConfigFile("logconfig.json", optional: true, reloadOnChange: true)
+
+					// Load all json files in the  `config` subdirectory (if it exists) of each configuration
+					// directory. We use the subdirectory to ensure that we only load configuration files.
+					.AddConfigFiles(subdirectory: "config", pattern: "*.json")
+
+					.AddEnvironmentVariables()
+					.AddCommandLine(args)
```

Comment:
I know this is the default priority, yet we should consider making the env vars take precedence.